### PR TITLE
added items in Readme Consensus

### PR DESCRIPTION
**Pull request by EdVeralli**
Description of the Consensus Algortihm

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -372,3 +372,166 @@ __Vf__ corresponds to field operations in a field __F__, and __Vg__ corresponds
...
+### Select chain
+Each time a peer's chains receive an update, the select chain event takes place.  
+A chain is said to be updated anytime a valid block is added or removed from its head. The chain selection algorithm also incorporates certain tiebreak logic.  
+Supplementary tiebreak logic becomes necessary when assessing chains with identical length or equal minimum density.
+
+Let ``P.tip`` refer to the top block of peer ``P``'s current best chain. Assuming an update to either ``P.tip`` or ``P.chains``, ``P`` must update its tip similar to this:
+![](/img/consensus06.png)
+
+The following selectSecureChain algorithm receives the peer's current best chain P.tip and its set of known valid chains P.chains and produces the most secure chain as output.  
+![](/img/consensus07.png)
```
Comment (xqft) [inline README.md:518]: Why images instead of code blocks?

File: README.md [hunk 2]
```diff
@@ -372,3 +372,166 @@ __Vf__ corresponds to field operations in a field __F__, and __Vg__ corresponds
...
+#### Decentralized checkpointing
+Samasika employs decentralized checkpointing to discern the nature of a fork, categorizing it as either short-range or long-range.
+- **Start checkpoint** - State hash of the first block of the epoch.
+- **Lock checkpoint** - State hash of the last known block in the seed update range of an epoch (not including the current block) 
+
+Remember, a fork is categorized as short-range if either:
+- The fork point of the candidate chains are in the same epoch.
+- The fork point is in the previous epoch with the same ``lock_checkpoint``
+
+As Mina prioritizes succinctness, it implies the need to maintain checkpoints for both the current and the previous epoch.
```
[the remark this answers is not in the document]
Comment (EdVeralli) [reply README.md:411]: I suppose I liked it too much to repeat it...:wink: I'm going to correct it

File: README.md (hunk 1 above)
Comment (EdVeralli) [reply README.md:517]: I wanted to keep the colors in the functions and instructions. It's just that.

File: README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (xqft) [reply README.md:411]: Thanks!

File: README.md (hunk 1 above)
Comment (xqft) [reply README.md:517]: I think that if you make a code block and set the proper language (Rust) it'll be colored. [Here's a little example](https://www.markdownguide.org/extended-syntax/#syntax-highlighting)

Comment (xqft) [reply README.md:517]: However it's not a big deal right now, we can open an issue about this!

**Review by xqft — APPROVED** [review]

**Pull request merged**